lollipop/types.py

```diff
@@ -1355,3 +1355,43 @@ def __repr__(self):
             klass=self.__class__.__name__,
             inner_type=repr(self.inner_type),
         )
+
+
+class Transform(Type):
+    """A wrapper type which allows us to convert data structures to an inner type,
+       then loaded or dumped with a customized format.
+    Example:
+
+        Point = namedtuple('Point', ['x', 'y'])
+
+        PointType = Transform(
+            Tuple(Integer(), Integer()),
+            load=lambda values: Point(values[0], values[1]),
+            dump=lambda point: [point.x, point.y],
+        )
+
+        PointType.dump((Point(x=1, y=2)))
+        # => [1,2]
+
+        PointType.load([1,2])
+        # => Point(x=1, y=2)
+
+    :param Type inner_type: Data type.
+    :param load_modifier: Specifies loaded format for object.
+    :param dump_modifier: Specifies dump format for object.
+
+    """
+    def __init__(self, inner_type, pre_load=None, post_load=None,
+                 pre_dump=None, post_dump=None):
+        super(Transform, self).__init__()
+        self.inner_type = inner_type
+        self.pre_load = pre_load or identity
+        self.post_load = post_load or identity
+        self.pre_dump = pre_dump or identity
+        self.post_dump = post_dump or identity
+
+    def load(self, data, *args, **kwargs):
+        return self.post_load(self.inner_type.load(self.pre_load(data), *args, **kwargs))
```

Comment:
You still need to account for cases when pre/post_load functions need to receive an extra "context" argument.